sources/gtests.cpp

```diff
@@ -0,0 +1,124 @@
+#include "extended_matrix.h"
+#include "gtest/gtest.h"
+#include <vector>
+
+
+TEST(matrix_general, assignment_concatenation) {
+
+    SparseMatrix<int, -1, 3> matrix;
```

Comment:
Did not find any test with `const SparseMatrix` and `const SparseMatrix&`? I think, it makes sense to be able to pass `const SparseMatrix` to function

Reply:
You are right, I missed this point.